server/src/main/java/org/elasticsearch/plugins/PluginDescriptor.java

```diff
@@ -119,7 +120,11 @@ public PluginDescriptor(final StreamInput in) throws IOException {
         this.version = in.readString();
         elasticsearchVersion = Version.readVersion(in);
         javaVersion = in.readString();
-        this.classname = in.readString();
```

Comment:
Question: do we have a policy of not throwing in constructors? (I guess not, since these read methods can throw) Or a preference to not do so?
Otherwise, I'd like to add some sanity checks (e.g. isStable <-> classname == null && moduleName == null), throwing IllegalArgumentException if not met.

Reply:
Throwing in constructors is fine, if the arguments are not valid

Reply:
I will add some checks in the constructors then. 
I also noticed that `toXContent` and `toString` are "outdated": they do not include all fields. I would update them, but I am wondering about `toXContent`. Is it used? A quick look seems to indicate it is used by `PluginsAndModules` and `PluginsAndModules` (for stats and report via NodeInfo).
I'd think it is safe to add the missing fields and omit classname/moduleName if null (if it is not already done by `XContentBuilder`!) 
What do you think?

Reply:
I would use a separate PR to add the missing fields - also maybe check the git history & previous PRs to see if they were deliberately omitted or not

Reply:
Sounds like a good plan, will do!